### PR TITLE
revises front-proxy sandbox to include envoy service sidecars alongsi…

### PR DESCRIPTION
**Pull request by eitansuez**
…de each service

Commit Message: revises front-proxy sandbox to include envoy service sidecars alongside each service
Additional Description: none
Risk Level: Low
Testing: the sandbox's verify.sh script
Docs Changes: none
Release Notes: revise front-proxy sandbox to include envoy service sidecars alongside each service
Platform Specific Features: none
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: examples/front-proxy/docker-compose.yaml [hunk 1]
```diff
@@ -18,13 +17,17 @@ services:
   service1:
     build:
       context: ../shared/python
-      target: aiohttp-tracing-service2
+      target: aiohttp-tracing-service3
+    volumes:
+    - ./service-envoy.yaml:/etc/service-envoy.yaml
```
Comment (phlax) [inline examples/front-proxy/docker-compose.yaml:22]: thanks for follow up @eitansuez 

im thinking we need to add separate envoy services for these sidecars

Comment (eitansuez) [reply examples/front-proxy/docker-compose.yaml:22]: both sidecar envoys listen on port 8000 and route to their local service listening on port 8080.
in what way would two distinct service-envoy.yaml's differ?

Comment (phlax) [reply examples/front-proxy/docker-compose.yaml:22]: i didnt mean that the sidecars require different config to each other

i meant that this is injecting the config into the backend service

Comment (phlax) [reply examples/front-proxy/docker-compose.yaml:22]: you can just update this PR - no need to create new one

Comment (eitansuez) [reply examples/front-proxy/docker-compose.yaml:22]: ok, got it. thanks for the clarification.

**Review by phlax — COMMENTED** [review]
looking good @eitansuez 

i think we also want to update the `front-proxy/depends_on` so its depending on the sidecars and not the backend services

File: examples/front-proxy/docker-compose.yaml [hunk 2]
```diff
@@ -15,6 +14,30 @@ services:
...
+      dockerfile: ../shared/envoy/Dockerfile
+      args:
+        ENVOY_CONFIG: ./service-envoy-1.yaml
+    depends_on:
+      service1:
+        condition: service_healthy
+    ports:
+    - "8011:8000"
+
+  service-envoy-2:
```
[the remark this answers is not in the document]
Comment (eitansuez) [reply examples/front-proxy/docker-compose.yaml:34]: done.

File: examples/front-proxy/docker-compose.yaml [hunk 3]
```diff
@@ -15,6 +14,30 @@ services:
...
+    build:
+      context: .
+      dockerfile: ../shared/envoy/Dockerfile
+      args:
+        ENVOY_CONFIG: ./service-envoy-1.yaml
+    depends_on:
+      service1:
+        condition: service_healthy
+    ports:
+    - "8011:8000"
```
[the remark this answers is not in the document]
Comment (eitansuez) [reply examples/front-proxy/docker-compose.yaml]: done.

File: examples/front-proxy/docker-compose.yaml [hunk 4]
```diff
@@ -15,6 +14,30 @@ services:
...
+    build:
+      context: .
+      dockerfile: ../shared/envoy/Dockerfile
+      args:
+        ENVOY_CONFIG: ./service-envoy-2.yaml
+    depends_on:
+      service2:
+        condition: service_healthy
+    ports:
+    - "8012:8000"
```
[the remark this answers is not in the document]
Comment (eitansuez) [reply examples/front-proxy/docker-compose.yaml]: done.

**Comment by eitansuez** [comment]
> we also want to update the front-proxy/depends_on so its depending on the sidecars and not the backend services

that would necessitate configuring healthchecks for the sidecars.  i'll see if i can find an example..

**Comment by phlax** [comment]
> that would necessitate configuring healthchecks for the sidecars. i'll see if i can find an example..

yeah - thinking further on this - i think we can make the front-proxy depend on the services being `service_healthy` and the sidecars being `service_started` - not 100% reliable, but i think probably will work (adding healthchecks would be better, im not sure we have done so in other Envoy services in our examples)

**Review by phlax — APPROVED** [review]
lgtm, really nice work -thanks @eitansuez !

**Comment by eitansuez** [comment]
really appreciate the guidance and responsiveness to help me figure out how to get this right.

**Pull request merged**